src/ILCompiler.Compiler/src/Compiler/PreInitFieldInfo.cs

```diff
@@ -25,11 +87,55 @@ public class PreInitFieldInfo
         /// </summary>
         public int Length { get; }
 
-        public PreInitFieldInfo(FieldDesc field, byte[] data, int length)
+        /// <summary>
+        /// List of fixup to be apply to the data blob
+        /// This is needed for information that can't be encoded into blob ahead of time before codegen
+        /// </summary>
+        public List<PreInitFixupInfo> FixupInfos { get; }
+
+        public PreInitFieldInfo(FieldDesc field, byte[] data, int length, List<PreInitFixupInfo> fixups)
         {
             Field = field;
             Data = data;
             Length = length;
+            FixupInfos = fixups;
+
+            if (FixupInfos != null)
+                FixupInfos.Sort();
+        }
+
+        public virtual int WriteData(ref ObjectDataBuilder builder, NodeFactory factory)
```

Comment:
There is no reason that I can identify for this method to be virtual, or for it to return anything.